### PR TITLE
Parallel Test Runs Part 2

### PR DESCRIPTION
**Pull request by SDESkowronski**
This is the conclusion of code changes intended to allow us to run Cypress test cases in parallel on Circle CI while maintaining the ability to locally run any combination of test cases in the order specified in TestList.js as the "testList" array.

**There is also a video to demonstrate these new features:**
[Demo of Adding and Removing Cypress Test Cases](https://microsoft.sharepoint.com/:v:/t/convsysresearch/EVVZz13Ql1FNmtXzbPYA0GUB_KHl0YOmY23UWzzARcVYLQ?e=xTpiW4)

### CODE REVIEW

File: cypress/GenerateCypressTestSpecs.js [hunk 1]
```diff
@@ -16,55 +16,82 @@
...
+    // console.log(testFileDirectory);
+    // console.log()
+
+    // Parse each test js file.
+    testFileDirectory.forEach(fileName => { ParseTestJsFile(fileName); });
+
+    // Finalize our findings by re-writing out the TestList.js file ONLY if it has changed.
+    var index = testList_jsContents.indexOf('// ************ Generated Code Beyond this Point *************************');
+    var newFileContents =
+      '// ************ Generated Code Beyond this Point *************************\r\n' +
```
Comment (mattmazzola) [inline cypress/GenerateCypressTestSpecs.js]: I had left comment in previous PR about using multiline strings. What was your opinion?
https://github.com/Microsoft/ConversationLearner-UI/pull/953#discussion_r251131899

File: cypress/support/TestListManager.js [hunk 2]
```diff
@@ -44,10 +44,22 @@ const train = require('../tests/Train');
...
+  // These two lines of code converts the testFunction from a callable
+  // function to a string containing the name of this function. It works
+  // like this:
+  //   1) We convert the function to a string containing the full function
+  //      that looks like: 
+  //        "function TheTestFunctionToRun(param1, param2) {...}"
+  //   2) The magic number 9 here is skipping past the string "function "
+  //      and grabbing just the name of the function.
+  var testFunctionAsString = testFunction.toString();
   var testFunctionName = testFunctionAsString.substring(9, testFunctionAsString.indexOf('(', 9));
```
Comment (mattmazzola) [inline cypress/support/TestListManager.js]: I had left another comment in previous PR about reducing magic constants. Also wanted to know opinions on this.
https://github.com/Microsoft/ConversationLearner-UI/pull/953#discussion_r251504923

File: cypress/GenerateCypressTestSpecs.js (hunk 1 above)
Comment (SDESkowronski) [reply cypress/GenerateCypressTestSpecs.js]: Liked the suggestion, made the fix.

File: cypress/support/TestListManager.js (hunk 2 above)
Comment (SDESkowronski) [reply cypress/support/TestListManager.js]: Liked part of the suggestion and implemented that part.

**Pull request merged**